modules/GKB/CommonUtils.pm

```diff
@@ -319,81 +379,76 @@ sub get_unique_species {
             } @unique_species_instances);
     }
-    
-    return sort { $a->displayName cmp $b->displayName } @unique_species_instances;    
+
+    my @sorted_unique_species_instances = sort { $a->displayName cmp $b->displayName } @unique_species_instances;
+    return @sorted_unique_species_instances;
 }
 
 sub XOR {
     my $expression1 = shift;
     my $expression2 = shift;
-    
+
     return ($expression1 || $expression2) && (!($expression1 && $expression2));
 }
 
-sub get_components
-{
-	my $composite_entity = shift;
-
-	my @components;
-	if ($composite_entity)
-	{
-		foreach my $component (@{$composite_entity->hasComponent}, @{$composite_entity->hasMember},
-								@{$composite_entity->hasCandidate}, @{$composite_entity->repeatedUnit})
-		{
-			push @components, $component;
-			my @sub_components = grep {defined} get_components($component);
-			if (@sub_components)
-			{
-				push @components, @sub_components;
-			}
-		}
-	}
-
-	return @components;
+sub get_components {
+    my $composite_entity = shift;
+
+    my @components;
+    if ($composite_entity) {
+        foreach my $component (
+            @{$composite_entity->hasComponent}, @{$composite_entity->hasMember},
+            @{$composite_entity->hasCandidate}, @{$composite_entity->repeatedUnit}
+        ) {
+            push @components, $component;
+            my @sub_components = grep {defined} get_components($component);
+            if (@sub_components) {
+                push @components, @sub_components;
+            }
+        }
+    }
+
+    return @components;
 }
 
 sub get_instance_creator {
     my $instance = shift;
-    
+
     if ($instance->created->[0] && $instance->created->[0]->author->[0]) {
         return $instance->created->[0]->author->[0]->displayName || 'Unknown';
     }
-    
+
     return 'Unknown';
 }
 
 sub get_instance_modifier {
     my $instance = shift;
-    
+
     return get_event_modifier($instance);
 }
 
 sub get_event_modifier {
-	my $event = shift;
-	
-	return 'Unknown' unless $event;
-	
-	my $author_instance;
-	try
-	{
-		foreach my $modified_instance (reverse @{$event->modified})
-		{
-			$author_instance ||= $modified_instance->author->[0] unless $modified_instance->author->[0] && $modified_instance->author->[0]->db_id == 140537;
-		}
-		$author_instance ||= $event->created->[0]->author->[0];
-	}
-	catch
-	{
-		confess "Error caught: $_ \nFor event: ".$event->extended_displayName;
-	}
-	my $author_name = $author_instance->displayName if $author_instance;
-	
-	return $author_name || 'Unknown';
+    my $event = shift;
+
+    return 'Unknown' unless $event;
+
+    my $author_instance;
+    try {
+        foreach my $modified_instance (reverse @{$event->modified}) {
+            $author_instance ||= $modified_instance->author->[0] unless $modified_instance->author->[0] && $modified_instance->author->[0]->db_id == 140537;
```

Comment:
What is the magical number?

Reply:
A magician never reveals his tricks...

I think it's Guanming's person id instance.  I'll work it!

Reply:
Maybe a better question is: why are things authored by Guanming being treated differently?

Reply:
Sorry, I should have explained.  

Guanming (and others like myself I may need to add in) are excluded because this function is meant to return the curator author and many modification instances are made by Guanming for data model purposes that Lisa requested be ignored.

I'll include a comment with the function and refactor to make the excluded authors more obvious.